stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/TypeAdapterGenerator.java

```diff
@@ -245,11 +246,6 @@ private static MethodSpec getReadMethodSpec(@NotNull TypeName typeName,
                         "();\n" +
                         "\twhile (reader.hasNext()) {\n" +
                         "\t\tString name = reader.nextName();\n" +
-                        "\t\tcom.google.gson.stream.JsonToken jsonToken = reader.peek();\n" +
-                        "\t\tif (jsonToken == com.google.gson.stream.JsonToken.NULL) {\n" +
-                        "\t\t\treader.skipValue();\n" +
-                        "\t\t\tcontinue;\n" +
-                        "\t\t}\n" +
```

Comment:
Primary reason is performance


1 - uninitialized fields should not be a problem as that can happen if the server does not send the data. Previous implementation also would have created an object with similar state.
2. Not if we ensure all primitive type adapters in KnownTypeAdapters are null safe. 

But with this you have given me an idea. I will implement methods in KnownTypeAdapters of the sort :

int read(JsonReader in, int default value)

Which will be called as :

obj.field = KnownTypeAdapters.read(reader, obj.field)

In the event of the response being null we will either set it to a default value or its existing value